services/repository/transfer.go

```diff
@@ -360,15 +365,20 @@ func ChangeRepositoryName(ctx context.Context, doer *user_model.User, repo *repo
 	oldRepoName := repo.Name
 
 	// Change repository directory name. We must lock the local copy of the
-	// repo so that we can atomically rename the repo path and updates the
+	// repo so that we can automatically rename the repo path and updates the
 	// local copy's origin accordingly.
 
-	repoWorkingPool.CheckIn(fmt.Sprint(repo.ID))
+	ctx, releaser, err := globallock.Lock(ctx, getRepoWorkingLockKey(repo.ID))
```

Comment:
A lot of such usages are wrong.

Many operations are not revert-able. If the lock ctx is used, then the instance goes corrupted.

Again, I do not think the "lock ctx" makes any real sense in real world.

Reply:
I don't think it's related to the lock `ctx` implementation but because the function `changeRepositoryName` implementation has defect. Even use the current implementation, when `if err = util.Rename(repo.RepoPath(), newRepoPath)` succeed, but database operation failed, then the repository can not be accessed.

Reply:
But it doesn't make sense to make the situation worse. I do not think a canceled lock context should stop the non-revert-able operations (while most operations are non-revert-able in current code base).